src/templates/ObjectTemplate.java

```diff
@@ -151,15 +152,15 @@ public class {{ impl_class_name }} implements Disposable, AutoCloseable, {{ inte
   // Note no constructor generated for this object as it is async.
   {%-     else %}
   {%- call java::docstring(cons, 4) %}
-  public {{ impl_class_name }}({% call java::arg_list(cons, true) -%}){
+  public {{ impl_class_name }}({% call java::arg_list(cons, true) -%}) {% match cons.throws_type() %}{% when Some(throwable) %}throws {{ throwable|type_name(ci) }}{% else %}{% endmatch %}{
     this((Pointer){%- call java::to_ffi_call(cons) -%});
   }
   {%-     endif %}
   {%- when None %}
   {%- endmatch %}
 
   @Override
-  public void destroy() {
+  public synchronized void close() {
```

Comment:
Since this is `close()` instead of `destroy()`, should it be `wasClosed` instead of `wasDestroyed`? Maybe `wasDestroyed` is fine, since it looks like it does destroy the fields when closing.

Reply:
Yeah it's a bit of both. `close` is the Java parlance and the Java interface we've implemented, but `destroy` is the common-across-bindgens uniffi word for it.